learn/cookbooks/docker.md

```diff
@@ -0,0 +1,135 @@
+# Using Meilisearch with Docker
+
+A common way of installing Meilisearch is using Docker. In this guide you will learn what Docker is, how to install it in your local machine, and how to use Meilisearch with it.
+
+## Docker
+
+Docker is a tool that packages an application's code and dependencies into something called a container. Using Docker allows developers to ensure their software runs reliably across a wide range of machines and environments.
+
+Docker containers are distributed in images. Docker images are packages storing not only the container, but also instructions on how Docker should instantiate that container.
+
+## Install Docker
+
+The first step to start using Docker is installing it in your target machine. We recommend following [Docker's official instructions to install Docker Desktop](https://docs.docker.com/get-docker/), available for MacOS, Windows, and Linux.
+
+Docker for Desktop includes an interface for managing your containers, command-line applications and several tools required for using Docker.
+
+## Install Meilisearch
+
+Once you installed Docker you can use the `docker pull` command to download a Meilisearch image:
+
+```sh
+docker pull getmeili/meilisearch:v0.28
+```
+
+Meilisearch deploys a new Docker image for every new release. All images are tagged with the release version. You can see [the full list of available Meilisearch Docker images](https://hub.docker.com/r/getmeili/meilisearch/tags#!) on Docker Hub.
+
+::: warning
+Meilisearch advises against using the `latest` tag. As its name indicates, `latest` will always get the most recent Meilisearch release, which might result in different machines running different images if significant time passes between setting up each of them.
+:::
+
+## Run Meilisearch
+
+After running `docker pull` and installing Meilisearch, you can use `docker run` to launch it:
+
+```sh
+docker run -it --rm \
+  -p 7700:7700 \
+  -v $(pwd)/meili_data:/meili_data \
+  getmeili/meilisearch:v0.28
+```
+
+### Passing instance options
+
+Meilisearch accepts a number of instance options during launch. You can configure instance options in two ways: environment variables and CLI arguments.
+
+#### Passing instance options with environment variables
+
+To pass environment variables to Docker, you must use the `-e` argument. The example below launches Meilisearch with a master key:
+
+```sh
+docker run -it --rm \
+  -p 7700:7700 \
+  -e MEILI_MASTER_KEY='MASTER_KEY'\
+  -v $(pwd)/meili_data:/meili_data \
+  getmeili/meilisearch:v0.28
+```
+
+#### Passing instance options with CLI arguments
+
+In the previous examples, Docker implicitly runs Meilisearch. If you want to pass command-line arguments to Meilisearch with Docker, you must explicitly run the `meilisearch` binary:
+
+```sh
+docker run -it --rm \
+  -p 7700:7700 \
+  -v $(pwd)/meili_data:/meili_data \
+  getmeili/meilisearch:v0.28
+  meilisearch --master-key="MASTER_KEY"
```

Comment:
For me, it's a bit confusing to use the same example as an environment variable and as a CLI command, I'd be the type to think that I can pass all the options to CLI-command

Reply:
Actually, I checked and it is almost always the case except for some commands so ... I don't  know if you deal with links but maybe we can have the link to the `env` variable page

Reply:
It's a good remark, I thought about it when I've first seen it, and then checked PostgreSQL (which is my ultimate reference of what a good dockerfile should be) and they showed both way to do it so I thought it was ok.
But after your comment I checked a bunch of other very well known docker image and it look like most people only show the env variable way;
- [PostgreSQL](https://hub.docker.com/_/postgres) show both way to do it
- [gogs](https://hub.docker.com/r/gogs/gogs) only uses env variables
- [gitlab](https://docs.gitlab.com/ee/install/docker.html) only env variables
- [deluge](https://hub.docker.com/r/linuxserver/deluge) only env variables
- [radarr](https://hub.docker.com/r/linuxserver/radarr) only env var
- [calibre](https://hub.docker.com/r/linuxserver/calibre) only env variables
- [plex](https://hub.docker.com/r/linuxserver/plex) only env var
- [sqlite](https://hub.docker.com/r/linuxserver/sqlitebrowser) only env var
- [LibreOffice](https://hub.docker.com/r/linuxserver/libreoffice) only env var
- [duckdns](https://hub.docker.com/r/linuxserver/duckdns) only env var
- [pihole](https://hub.docker.com/r/pihole/pihole) only env var but almost no variables
- [nginx](https://hub.docker.com/_/nginx/) use a config file or env variables since the last version
- [apache](https://hub.docker.com/_/httpd) use a config file

Reply:
That's interesting, thanks for the references. 

I dislike having two very similar ways of configuring an instance with Docker. It just drops the decision on the user's lap: "both ways are pretty much the same and have similar results, flip a coin or something if you can't decide."

However, despite what other docs do, I think it's very important to show the CLI method because there are certain Meilisearch options (e.g. `--import-dump`, `--import-snapshot`, and their siblings) that are not available as env vars.